pyface/data_view/abstract_data_model.py

```diff
@@ -0,0 +1,272 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Provides an AbstractDataModel ABC for Pyface data models.
+
+This module provides an ABC for all data view data models.  This specifies
+the API that the data view widgets expect, and which the underlying
+data is adapted to by the concrete implementations.  Data models are intended
+to be toolkit-independent, and be able to adapt any approximately tabular or
+nested data structure to what the data view system expects.
+"""
+from abc import abstractmethod
+
+from traits.api import ABCHasStrictTraits, Event, Instance
+
+from .index_manager import AbstractIndexManager
+
+
+class AbstractDataModel(ABCHasStrictTraits):
+    """ Abstract base class for Pyface data models.
+
+    The data model API is intended to provide a common API for heirarchical
+    and tabular data.  This class is concerned with the structure, type and
+    values provided by the data, but not with how the data is presented.
```

Comment:
It's not "business logic", because this is all UI code, but it is pretty much the model part of these patterns.  It's not a controller, viewmodel or a presenter, because it doesn't get user events.

And I chose "presented" because it's not only says nothing about "display" but also nothing about choice of editors, selection modes, etc.